lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -1747,9 +1747,26 @@ bool AssociatedTypeInference::diagnoseNoSolutions(
           if (failed.Result.isError())
             continue;
 
+          if (!failed.TypeWitness->is<NominalType>() &&
```

Comment:
Wait, I'm confused. I think this is ok because the test is negative. An existential can't conform to a protocol and an existential is not a nominal. A protocol can conform, and a protocol is a nominal. Do I have part of this wrong?